File: tests/conformance/fixtures/typescript/04-gate.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('build').run('make build');

p.gate('approve-deploy').after('build')
  .gateStrategy('env')
  .gateTimeout(600)
  .gateMessage('Approve deployment to production?')
  .gateEnvVar('DEPLOY_APPROVED');

p.task('deploy').run('make deploy').after('approve-deploy');

p.emit();

File: tests/conformance/fixtures/go/04-gate.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("build").Run("make build")

	p.Gate("approve-deploy").After("build").
		GateStrategy("env").
		GateTimeout(600).
		GateMessage("Approve deployment to production?").
		GateEnvVar("DEPLOY_APPROVED")

	p.Task("deploy").Run("make deploy").After("approve-deploy")

	p.Emit()
}
